lib/ovirt/service.rb

```diff
@@ -127,11 +128,13 @@ def status(link)
     end
 
     def api_uri(path = nil)
-      uri = "#{base_uri}/api"
+      uri = "#{base_uri}"
+      uri += api_version == 'v4' ? '/ovirt-engine/api' : '/api'
       unless path.nil?
         parts = path.to_s.split('/')
-        parts.shift if parts.first == ''    # Remove leading slash
-        parts.shift if parts.first == 'api' # We already have /api in our URI
+        parts.shift if parts.first == ''             # Remove leading slash
+        parts.shift if parts.first == 'ovirt-engine' # We already have /ovirt-engine in our URI
+        parts.shift if parts.first == 'api'          # We already have /api in our URI
         uri += "/#{parts.join('/')}" unless parts.empty?
```

Comment:
:+1: string building causes all sorts of problems with IPv6.  base_uri should provide a URI object as it's name implies.

http://ruby-doc.org/stdlib-2.3.0/libdoc/uri/rdoc/URI.html